File: Examples/Simple Integration Obj-C/Source/ProductViewCell.m
//
//  CollectionViewCell.m
//  Simple Integration Obj-C
//
//  Created by Johnny Peter on 29/08/19.
//  Copyright © 2019 Network International. All rights reserved.
//

#import "ProductViewCell.h"
#import "Simple_Integration_Obj_C-Swift.h"

#define MIN_PRICE 10
#define MAX_PRICE 50

@interface ProductViewCell ()
@end

@implementation ProductViewCell

@synthesize productLabel;
@synthesize priceLabel;

- (id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if(self) {
        self.priceLabel = [[UILabel alloc] init];
        self.productLabel = [[UILabel alloc] init];
        self.price = MIN_PRICE + arc4random_uniform((uint32_t)(MAX_PRICE - MIN_PRICE + 1));
        self.priceLabel.text = [NSString stringWithFormat: @"AED %d", self.price];
        [self addSubViews];
    }
    return self;
}

- (void)addSubViews {
    self.productLabel.font = [self.productLabel.font fontWithSize:50];
    UIStackView *vStack = [[UIStackView alloc] init];
    [vStack addArrangedSubview: self.productLabel];
    [vStack addArrangedSubview: self.priceLabel];
    vStack.axis = UILayoutConstraintAxisVertical;
    vStack.alignment = UIStackViewAlignmentCenter;
    [self.contentView addSubview: vStack];
    
    [vStack anchorWithTop: self.contentView.topAnchor leading:self.contentView.leadingAnchor bottom:self.contentView.bottomAnchor trailing:self.contentView.trailingAnchor padding:UIEdgeInsetsMake(20, 20, 20, 20) size:CGSizeZero];
    
    self.contentView.layer.cornerRadius = 5;
    self.contentView.layer.masksToBounds = true;
    [self updateBorder: false];
}

- (void)updateBorder:(Boolean)selected {
    UIColor *borderColor = selected
    ? [UIColor colorWithRed:0.40 green:0.73 blue:0.40 alpha:1.0]
    : [UIColor colorWithRed:0.98 green:0.98 blue:0.98 alpha:1.0];
    
    [self updateWith:borderColor];
}

- (void)updateWith:(UIColor*)borderColor {
    [self.contentView addBorder: UIRectEdgeTop color: borderColor thickness: 5];
    [self.contentView addBorder: UIRectEdgeBottom color: borderColor thickness: 5];
    [self.contentView addBorder: UIRectEdgeLeft color: borderColor thickness: 5];
    [self.contentView addBorder: UIRectEdgeRight color: borderColor thickness: 5];
}

@end

File: Examples/Simple Integration/Source/ProductViewCell.swift
//
//  ProductViewCell.swift
//  Simple Integration
//
//  Created by Johnny Peter on 22/08/19.
//  Copyright © 2019 Network International. All rights reserved.
//

import Foundation
import UIKit

class ProductViewCell: UICollectionViewCell {
    let productLabel = UILabel()
    let priceLabel = UILabel()
    var price: Double = 0

    private static let niBlue = UIColor(red: 0.0/255.0, green: 85.0/255.0, blue: 222.0/255.0, alpha: 1.0)

    override init(frame: CGRect) {
        super.init(frame: frame)
        addViews()
    }

    func updateBorder(selected: Bool) {
        if selected {
            contentView.layer.borderColor = ProductViewCell.niBlue.cgColor
            contentView.layer.borderWidth = 2
            contentView.backgroundColor = ProductViewCell.niBlue.withAlphaComponent(0.05)
        } else {
            contentView.layer.borderColor = UIColor.separator.cgColor
            contentView.layer.borderWidth = 1
            contentView.backgroundColor = UIColor.secondarySystemBackground
        }
    }

    func setProduct(product: Product) {
        productLabel.text = product.name
        price = product.amount
        priceLabel.text = String(format: "%.2f", price)
        accessibilityIdentifier = "product_cell_\(product.name.lowercased().replacingOccurrences(of: " ", with: "_"))"
    }

    func addViews() {
        contentView.layer.cornerRadius = 12
        contentView.layer.masksToBounds = true
        contentView.backgroundColor = UIColor.secondarySystemBackground

        productLabel.font = UIFont.systemFont(ofSize: 13, weight: .medium)
        productLabel.textColor = .secondaryLabel
        productLabel.textAlignment = .center
        productLabel.accessibilityIdentifier = "product_label_name"

        priceLabel.font = UIFont.systemFont(ofSize: 22, weight: .bold)
        priceLabel.textColor = .label
        priceLabel.textAlignment = .center
        priceLabel.accessibilityIdentifier = "product_label_price"

        let vStack = UIStackView(arrangedSubviews: [priceLabel, productLabel])
        vStack.axis = .vertical
        vStack.alignment = .center
        vStack.spacing = 4
        contentView.addSubview(vStack)

        updateBorder(selected: false)

        vStack.anchor(top: contentView.topAnchor, leading: contentView.leadingAnchor,
                     bottom: contentView.bottomAnchor, trailing: contentView.trailingAnchor,
                     padding: UIEdgeInsets(top: 16, left: 12, bottom: 16, right: 12),
                     size: .zero)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
